src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -2906,7 +2910,11 @@ private void onApplicationWindow(
                 if (subscription != null &&
                         !subscription.hasSubscribeCompleted(subackIndex) && hasSubscribeCapability(capabilities))
                 {
-                    subscription.onSubscribeSucceeded(traceId, authorization, packetId, subackIndex);
+                    final boolean completed = subscription.onSubscribeSucceeded(traceId, authorization, packetId, subackIndex);
+                    if (completed)
```

Comment:
Avoid chaining the `boolean` `completed` flag through the methods and instead use the new `subscription.acknowledged()` method for the `if` condition.